**solution of ex 2.1**\
$\\ \textbf{optimisation problem as linear problem}\ \\ max\ 0.082x_1+0.036x_2+0.0156x_3+0.033x_4+0.054x_5+0.0125y \ \\ 3x_1-3x_2+5x_3-x_4+x_5-y <=80 \\ \ x_3+2x_4+4_y >=30 \\ \ 3x_1+ 3.6x_2-1.5x_3+5.6x_4-1.8x_5+6x_6 <=0 \\ \ 7x_1-2x_2-3x_3+4x_4+5x_5-6x_6+12y<=0 \\ \ x_1,x_2,x_3,x_4,x_5,x_6>=0 \\ \ -0.2<=y<=5.5 \\ \ assumption \\ \ -x_2-x_4+x_5 >0 \\ \ and -x_1+x_3+x_6-2y>0$

In [1]:
!pip install -q pyomo

     |████████████████████████████████| 9.6 MB 8.0 MB/s 
     |████████████████████████████████| 49 kB 5.3 MB/s 


In [2]:
from pyomo.environ import *

In [3]:
model2 =ConcreteModel()

In [4]:
#declare variable
model2.x1 = Var(within=NonNegativeReals)
model2.x2 =Var(within=NonNegativeReals)
model2.x3 =Var(within=NonNegativeReals)
model2.x4 =Var(within=NonNegativeReals)
model2.x5 =Var(within=NonNegativeReals)
model2.x6 =Var(within=NonNegativeReals)
model2.y =Var(bounds=(-0.2,5.5))

In [5]:
#declare objctive
model2.cost=Objective(expr= 0.082*model2.x1 + 0.036*model2.x2 + 0.0156*model2.x3 + 0.033*model2.x4 + 0.054*model2.x5 + 0.0125*model2.y,sense=maximize)

In [6]:
#declare constraints
model2.con1=Constraint(expr= 3*model2.x1 - 3*model2.x2 + 5*model2.x3 - model2.x4 + model2.x5 - model2.y <= 80)
model2.con2=Constraint(expr= model2.x3 + 2*model2.x4 + 4*model2.y >=30)
model2.con3 = Constraint(expr = 3*model2.x1 + 3.6*model2.x2 - 1.5*model2.x3 + 5.6*model2.x4 - 1.8*model2.x5 + 6*model2.x6 <= 0)
model2.con4 = Constraint(expr = 7*model2.x1 - 2*model2.x2 - 3*model2.x3 + 4*model2.x4 + 5*model2.x5 - 6*model2.x6 + 12*model2.y <= 0)

In [7]:
!apt-get install -y -qq glpk.utils

Selecting previously unselected package libsuitesparseconfig5:amd64.
(Reading database ... 155673 files and directories currently installed.)
Preparing to unpack .../libsuitesparseconfig5_1%3a5.1.2-2_amd64.deb ...
Unpacking libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libamd2:amd64.
Preparing to unpack .../libamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libcolamd2:amd64.
Preparing to unpack .../libcolamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libcolamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libglpk40:amd64.
Preparing to unpack .../libglpk40_4.65-1_amd64.deb ...
Unpacking libglpk40:amd64 (4.65-1) ...
Selecting previously unselected package glpk-utils.
Preparing to unpack .../glpk-utils_4.65-1_amd64.deb ...
Unpacking glpk-utils (4.65-1) ...
Setting up libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Setting up libcolamd2:amd64 (1:5.1.2-2) ...
Setting up libamd2:amd64 

In [8]:
model2.pprint()

7 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :     0 :  None :  None : False :  True : NonNegativeReals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :     0 :  None :  None : False :  True : NonNegativeReals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :     0 :  None :  None : False :  True : NonNegativeReals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :     0 :  None :  None : False :  True : NonNegativeReals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :     0 :  None :  None : False :  True : NonNegativeReals
    x6 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :     0 :  None :  None : False :  True : NonNegativeReals
    y : S

In [9]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model2).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 2.12102483443709
  Upper bound: 2.12102483443709
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 8
  Number of nonzeros: 23
  Sense: maximize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.010277032852172852
# ----------------------------------------------------------
#   Solution Information
# -------------

In [10]:
# display solution
print('\nCost = ', model2.cost())

print('\nDecision Variables')
print('x1 = ', model2.x1())
print('x2 = ', model2.x2())
print('x3 = ', model2.x3())
print('x4 = ', model2.x4())
print('x5 = ',model2.x5())
print('y = ',model2.y())

print('\nConstraints')
print('Constraint 1  = ', model2.con1())
print('Constraint 2  = ', model2.con2())
print('Constraint 3  = ', model2.con3())
print('Constraint 4  = ', model2.con4())


Cost =  2.1210248344370837

Decision Variables
x1 =  0.0
x2 =  19.5529801324503
x3 =  24.2384105960265
x4 =  0.0
x5 =  18.9072847682119
y =  1.44039735099338

Constraints
Constraint 1  =  80.00000000000011
Constraint 2  =  30.00000000000002
Constraint 3  =  -9.237055564881302e-14
Constraint 4  =  -6.039613253960852e-14


$\\ \ **solution \ \ \ \ \ of \ \ \ \ 2.3** \\ \textbf{optimal solution value = 2.12102483} \\ \textbf{variables value} \\ \ x_1=0 \\ \ x_2 = 19.552980 \\ \ x_3 = 24.23841059 \\ \ x_4 = 0 \\ \ x_5 = 18.9072847682119 \\ \ y= 1.44039735099 \\ \ consraint \ \ 1 \ \ \ is \ \ \  active \\ \ consraint  \ \ \ \ 2 \ \ \ \ is \ \ \ \  active \\ \ constraint \ \ \ \ 3 \ \ \ \ is \ \ \ \ inactive \\ \ consraint \ \ \ \ 4 \ \ \ \ is \ \ \ \ inactive$

$ \\ \textbf{solution of 2.4}  \\ \ my \ \ \ assumption \ \ \ is \ \ \ denominator \ \ \ must \ \ \ be \ \ \ greater \ \ \ than \ \ \ 0$